pkgs/applications/misc/insulator2/default.nix

```diff
@@ -20,50 +21,65 @@
 , rustc
 , jq
 , moreutils
+, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
   pname = "insulator2";
-  version = "2.12.2";
+  version = "2.13.2";
 
   src = fetchFromGitHub {
     owner = "andrewinci";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-Bi9GCQr7yox5Plc7o0svRKYi1XoK/HDGj1VbW1z4jac=";
+    hash = "sha256-34JRIB7/x7miReWOxR/m+atjfUiE3XGyh9OBSbMg3m4=";
   };
 
+  patches = [
+    # see: https://github.com/andrewinci/insulator2/pull/733
+    (fetchpatch {
+      name = "fix-rust-1.80.0";
+      url = "https://github.com/andrewinci/insulator2/commit/7dbff0777c4364eec68cf90488d99f06b11dfa98.patch";
+      hash = "sha256-P8rBufYpREP5tOO9vSymvms4f2JbsUEjK7/yn9P7gYk=";
+    })
```

Comment:
I'd prefer we wait for this to have an actual in-tree commit, or just vendor it here

Reply:
i submitted one week ago: https://github.com/andrewinci/insulator2/pull/733

i guess this will take much time...

Reply:
I'd recommend vendoring the patch here then. It's just more reliable than a commit from a fork

Reply:
this is already the official repo... the fork would be https://github.com/Stunkymonkey/insulator2

Reply:
Commit from a GitHub PR isn't guaranteed to be stable. Upstream author may require you to do a force push which may break the visibility of this commit hash. Thus it would be better to add a patch in this repo instead.

And since you already have Cargo.lock here, you can patch Cargo.toml only, and copy Cargo.lock